packages/atomic/src/components/search/result-lists/atomic-result-list/atomic-result-list.tsx

```diff
@@ -164,7 +201,26 @@ export class AtomicResultList implements InitializableComponent {
     });
   }
 
+  @Watch('tabManagerState')
+  watchTabManagerState(
+    newValue: {activeTab: string},
+    oldValue: {activeTab: string}
+  ) {
+    if (newValue?.activeTab !== oldValue?.activeTab) {
+      this.bindings.store.unsetLoadingFlag(this.loadingFlag);
+    }
+  }
+
   public render() {
+    if (
+      !shouldDisplayOnCurrentTab(
+        [...this.tabsIncluded],
+        [...this.tabsExcluded],
+        this.tabManagerState.activeTab
+      )
+    ) {
+      return <Hidden></Hidden>;
+    }
```

Comment:
Wondering if this should be a `TabGuard` or something 🤔

Reply:
Good idea :)